spec/inferno/repositories/validator_sessions_spec.rb

```diff
@@ -0,0 +1,156 @@
+  let(:repo) { described_class.new }
+  let(:validator_name) { 'basic_name' }
+  let(:test_suite_id) { 'basic_suite' }
+  let(:validator_session_id1) { 'basic_validator1' }
+  let(:validator_session_id2) { 'basic_validator2' }
+  let(:validator_session_id3) { 'basic_validator3' }
+  let(:suite_options1) { { ig_version: '1', us_core_version: '4' } }
+  let(:suite_options2) { { ig_version: '2' } }
+  let(:suite_options_alt) { { us_core_version: '4', ig_version: '1' } }
+  let(:session1_params) do
+    {
+      validator_session_id: validator_session_id1,
+      validator_name:,
+      test_suite_id:,
+      suite_options: suite_options1
+    }
+  end
+  let(:session2_params) do
+    {
+      validator_session_id: validator_session_id2,
+      validator_name:,
+      test_suite_id:,
+      suite_options: suite_options2
+    }
+  end
+  let(:session3_params) do
+    {
+      validator_session_id: validator_session_id3,
+      validator_name:,
+      test_suite_id:,
+      suite_options: suite_options1
+    }
+  end
+  let(:session_params_alt1) do
+    {
+      validator_session_id: validator_session_id2,
+      validator_name:,
+      test_suite_id:,
+      suite_options: suite_options_alt
+    }
+  end
+  let(:session4_params) do
+    {
+      validator_session_id: validator_session_id2,
+      validator_name: 'alt name',
+      test_suite_id:,
+      suite_options: suite_options1
+    }
+  end
+  let(:session5_params) do
+    {
+      validator_session_id: validator_session_id2,
+      validator_name:,
+      test_suite_id: 'alt id',
+      suite_options: suite_options1
+    }
+  end
+
+  describe '#create' do
+    before do
+      repo.save(session1_params)
+    end
+
+    context 'with valid params' do
+      it 'persists data' do
+        record = repo.db.first
+        expect(repo.db.count).to eq(1)
+        expect(record[:validator_session_id]).to eq(validator_session_id1)
+        expect(record[:validator_name]).to eq(validator_name)
+        expect(record[:test_suite_id]).to eq(test_suite_id)
+      end
+
+      it 'creates a separate record given a different validator session id' do
+        repo.save(session2_params)
+        record = repo.db.all[1]
+        expect(repo.db.count).to eq(2)
+        expect(record[:validator_session_id]).to eq(validator_session_id2)
+        expect(record[:validator_name]).to eq(validator_name)
+        expect(record[:test_suite_id]).to eq(test_suite_id)
+      end
+
+      it 'overwrites an existing record given the same validator session id' do
+        repo.save(session3_params)
+        record = repo.db.first
+        expect(repo.db.count).to eq(1)
+        expect(record[:validator_session_id]).to eq(validator_session_id3)
+        expect(record[:validator_name]).to eq('basic_name')
+        expect(record[:test_suite_id]).to eq(test_suite_id)
+      end
+    end
+  end
+
+  describe '#find' do
+    before do
+      repo.save(session1_params)
+    end
+
+    it 'single record' do
```

Comment:
The descriptions on these `it` calls don't make sense. They should read like the ones above. I think this is good to go otherwise.

Reply:
updated